homework-solutions/TS/hw-13/task-2.ts

```diff
@@ -0,0 +1,29 @@
+/*
+
+Создайте функцию validatePassword, которая принимает строку (пароль) и возвращает true, 
+если пароль соответствует следующим правилам:
+  - Пароль должен содержать хотя бы одну заглавную букву.+
+  - Пароль должен содержать хотя бы одну букву в нижнем регистре. + 
+  - Пароль должен содержать хотя бы одну цифру. + 
+  - Пароль должен быть не менее 8 символов.+
+  - Пароль не должен состоять из одних пробелов +
+Функция должна возвращать false, если хотя бы одно из условий не выполнено.
+ */
+
+
+function validatePassword (password: string): boolean {
+    if (password.length < 8 || (password.trim().length === 0) || (!/\d/.test(password)) || (!/[a-zа-яё]/.test(password)) || (!/[A-ZА-ЯЁ]/.test(password))) {
+        return false
+    } 
+    
+    return true
+    
```

Comment:
Можно немного поменять проверки и возвращать полученное булевое значение
```suggestion
    return ( password.length >= 8 &&
     password.trim().length !== 0 &&
      /\d/.test(password) &&
       /[a-zа-яё]/.test(password) &&
       /[A-ZА-ЯЁ]/.test(password)) 
```